test/e2e/operator_test.go

```diff
@@ -350,3 +355,129 @@ func customResolver(nameserver string) *net.Resolver {
+			},
+			Spec: routev1.RouteSpec{
+				Host: host,
+				Path: path,
+				To: routev1.RouteTargetReference{
+					Kind: "Service",
+					Name: "testServiceName",
+				},
+			},
+		}
+	}
+	// use unique names for each test route to simplify debugging.
+	routeName := types.NamespacedName{Namespace: testIngressNamespace, Name: "external-dns-route"}
+	route := makeRoute(routeName, "app."+hostedZoneDomain, "/apis")
+
+	// The first route should be admitted
+	if err = kubeClient.Create(context.TODO(), route); err != nil && !errors.IsAlreadyExists(err) {
+		t.Fatalf("failed to create route: %v", err)
+	}
+	defer kubeClient.Delete(context.TODO(), route)
+	canonicalName := ""
+	t.Logf("Getting canonicalName for the route :%s ", routeName.Name)
+	if canonicalName, err = fetchRouterCanonicalHostname(t, routeName); err != nil {
+		t.Fatalf("Failed to get RouterCanonicalHostname for route %s/%s: %v", routeName.Namespace, routeName.Name, err)
+	}
+	t.Logf("canonicalName  : %s for the route :%s ", routeName.Name, canonicalName)
+	verifyOpenShiftRouteSource(t, canonicalName, fmt.Sprintf("app.%s", hostedZoneDomain))
+}
+
+func verifyOpenShiftRouteSource(t *testing.T, canonicalName, host string) {
+	//var canonicalName = "router-external-dns.external-dns.example-naga.info"
+	// try all nameservers and fail only if all failed
+	recordExist := false
+	for _, nameSrv := range nameServers {
+		t.Logf("Looking for cname record in nameserver: %s", nameSrv)
+		if err := wait.PollImmediate(dnsPollingInterval, dnsPollingTimeout, func() (done bool, err error) {
+			cname, err := lookupCNAME(host, nameSrv)
+			if err != nil {
+				t.Logf("cname lookup failed for nameserver : %s , error : %v", nameSrv, err)
+				return false, nil
+			}
+			if strings.Contains(cname, canonicalName) {
+				recordExist = true
+				return true, nil
+			}
+			return false, nil
+		}); err != nil {
+			t.Logf("Failed to verify host record with CNAME Record")
+		}
+	}
+
+	if !recordExist {
+		t.Logf("Cname record not found in any nameService, heance test failed")
+		t.Fail()
+		return
+	}
+}
+
+func fetchRouterCanonicalHostname(t *testing.T, route1Name types.NamespacedName) (string, error) {
+	customeRoute := routev1.Route{}
+	canonicalName := ""
+	if err := wait.PollImmediate(dnsPollingInterval, dnsPollingTimeout, func() (done bool, err error) {
+		err = kubeClient.Get(context.TODO(), types.NamespacedName{
+			Namespace: route1Name.Namespace,
+			Name:      route1Name.Name,
+		}, &customeRoute)
+		if err != nil {
+			return false, err
+		}
+		if len(customeRoute.Status.Ingress) < 1 {
+			t.Logf("No ingress found in route, retrying..")
+			return false, nil
+		}
+
+		for _, ingress := range customeRoute.Status.Ingress {
+			if strings.Contains(ingress.RouterCanonicalHostname, hostedZoneDomain) {
+				canonicalName = ingress.RouterCanonicalHostname
+			}
+		}
+		if canonicalName == "" {
+			t.Logf("unable to fetch the canonicalHostname, retrying..")
+			return false, nil
+		}
+		return true, nil
```

Comment:
```suggestion
	        t.Logf("No matching canonicalHostname, retrying..")
		return false, nil
```

Reply:
taken care already :)

Reply:
No, I still can see the check for the empty canonicalName. It's not needed anymore as we return the name is found.

Reply:
it should be like this.
```
	for _, ingress := range route.Status.Ingress {
			if strings.Contains(ingress.RouterCanonicalHostname, hostedZoneDomain) {
				canonicalName = ingress.RouterCanonicalHostname
				return true, nil
			}
		}
	return false, nil
```